x-pack/legacy/plugins/ml/public/application/data_frame_analytics/common/analytics.ts

```diff
@@ -341,3 +394,53 @@ export const loadEvalData = async ({
     return results;
   }
 };
+
+interface TrackTotalHitsSearchResponse {
+  hits: {
+    total: {
+      value: number;
+      relation: string;
+    };
+    hits: any[];
+  };
+}
+
+export const loadDocsCount = async ({
+  ignoreDefaultQuery = true,
+  isTraining,
+  searchQuery,
+  resultsField,
+  destIndex,
+}: {
+  ignoreDefaultQuery?: boolean;
+  isTraining: boolean;
+  searchQuery: SavedSearchQuery;
+  resultsField: string;
+  destIndex: string;
+}): Promise<{
+  docsCount: number | null;
+  success: boolean;
+}> => {
+  const query = getEvalQueryBody({ resultsField, isTraining, ignoreDefaultQuery, searchQuery });
+
+  try {
+    const body: SearchQuery = {
+      track_total_hits: true,
+      query,
+    };
+
+    const resp: TrackTotalHitsSearchResponse = await ml.esSearch({
+      index: destIndex,
+      size: 0,
+      body,
+    });
+
+    const docsCount = resp.hits.total && resp.hits.total.value;
```

Comment:
What happens if `total` is `undefined`? Do we want to return that with `success` still `true`?

Reply:
Good point - changing success to be set to `docsCount !== undefined` 👍